pep-9999.txt

```diff
@@ -0,0 +1,171 @@
+PEP: 9999
+Title: Python local packages directory [We need to find a proper name]
+Version: $Revision$
+Last-Modified: $Date$
+Author: Kushal Das <mail@kushaldas.in>, Steve Dower <steve.dower@python.org>,
+        Donald Stufft <donald@stufft.io>, Nick Coghlan <ncoghlan@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 16-May-2018
+Python-Version: 3.8
+
+
+
+Abstract
+========
+
+This PEP proposes to add to Python a mechanism to have a ``__pypackages__``
+directory  per project to have its own independent set of installed packages.
+This will share the standard library with the base installed Python. This will
+avoid the steps to create or activate or deactivate "virtual environments". Any
+Python scripts will use the ``__pypackages__`` from the base directory of the 
+script.
+
```

Comment:
Should probably add a sentence here along the lines of: "`pip` will also be updated to detect `__pypackages__` and install packages there by default" (and I think it ought to grow a `--local` option that will create `__pypackages__` if it doesn't exist)

Reply:
Hopefully this isn't *too* pedantic, but how exactly pip decides to implement this PEP isn't (and shouldn't) really be a question for this PEP. It can make recommendations, but it'll ultimately be up to the pip developers.

That being said, we'd probably add something like ``--local/user/global`` and ideally at some point default to ``--local``.

Reply:
Is there a better place to coordinate combined efforts between Python and pip?

That said, shouldn't we give recommendations to *Python package installers* in general, not just pip?

Reply:
Yea, we should be giving recommendations to the general installer (of which pip is one) and not just pip.

The PyPA is sorting out some organizational stuff at the moment, but I can forward this PR to folks there for anyone who cares to comment on it if we want wider feedback from PyPA members.